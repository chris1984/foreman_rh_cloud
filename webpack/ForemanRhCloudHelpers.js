/**
 * copied from core, since it's not in the ReactApp folder,
 * it's complicated to import it and mock it in tests.
 * should be imported once core moves it to the ReactApp folder.
 */
export const foremanUrl = path => `${window.URL_PREFIX}${path}`;

export const isNotRhelHost = ({ hostDetails }) =>
  // This regex tries matches sane variations of "RedHat", "RHEL" and "RHCOS"
  !new RegExp('red[\\s\\-]?hat|rh[\\s\\-]?el|rhc[\\s\\-]?os', 'i').test(
    // eslint-disable-next-line camelcase
    hostDetails?.operatingsystem_name
  );

export const vulnerabilityDisabled = ({ hostDetails }) =>
  isNotRhelHost({ hostDetails }) || !hostDetails?.vulnerability?.enabled;
