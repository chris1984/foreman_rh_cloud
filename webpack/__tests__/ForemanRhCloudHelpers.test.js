import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import { foremanUrl, vulnerabilityDisabled } from '../ForemanRhCloudHelpers';

global.URL_PREFIX = 'MY_TEST_URL_PREFIX.example.com';

const fixtures = {
  'should return foreman Url': () => foremanUrl('/test_path'),
  'vulnerabilityDisabled returns false for RHEL host with vulnerability enabled': () =>
    vulnerabilityDisabled({
      hostDetails: {
        operatingsystem_name: 'Red Hat Enterprise Linux',
        vulnerability: { enabled: true },
      },
    }),
  'vulnerabilityDisabled returns true for non-RHEL host': () =>
    vulnerabilityDisabled({
      hostDetails: {
        operatingsystem_name: 'Ubuntu',
        vulnerability: { enabled: true },
      },
    }),
  'vulnerabilityDisabled returns true for RHEL host with vulnerability disabled': () =>
    vulnerabilityDisabled({
      hostDetails: {
        operatingsystem_name: 'Red Hat Enterprise Linux',
        vulnerability: { enabled: false },
      },
    }),
  'vulnerabilityDisabled returns true for missing vulnerability object': () =>
    vulnerabilityDisabled({
      hostDetails: {
        operatingsystem_name: 'Red Hat Enterprise Linux',
      },
    }),
  'vulnerabilityDisabled returns true for missing hostDetails': () =>
    vulnerabilityDisabled({}),
};

describe('ForemanRhCloud helpers', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
