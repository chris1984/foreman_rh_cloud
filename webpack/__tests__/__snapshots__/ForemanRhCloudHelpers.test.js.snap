// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanRhCloud helpers should return foreman Url 1`] = `"MY_TEST_URL_PREFIX.example.com/test_path"`;

exports[`ForemanRhCloud helpers vulnerabilityDisabled returns false for RHEL host with vulnerability enabled 1`] = `false`;

exports[`ForemanRhCloud helpers vulnerabilityDisabled returns true for RHEL host with vulnerability disabled 1`] = `true`;

exports[`ForemanRhCloud helpers vulnerabilityDisabled returns true for missing hostDetails 1`] = `true`;

exports[`ForemanRhCloud helpers vulnerabilityDisabled returns true for missing vulnerability object 1`] = `true`;

exports[`ForemanRhCloud helpers vulnerabilityDisabled returns true for non-RHEL host 1`] = `true`;
